calculator-app/src/App.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import './App.css';
+import Button from './components/Button';
+import Input from './components/Input'
+import ClearButton from './components/ClearButton'
+import * as math from 'mathjs'
+
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      input: ""
+    }
+  }
+
+  updateInput = (val) => {
+    this.setState({ 
+      input: this.state.input + val});
```

Comment:
I see what you are doing in here, but this actually creates a lot of problems on your app. Because of this I'm able to do "+++---***" on the screen,  or “Infinity000+++“, or “2..3…4…“, etc, as it keeps adding to the screen.